sambamba/sort.d

```diff
@@ -154,7 +173,13 @@ class Sorter {
                 auto len = read.raw_data.length;
                 assert(len > max_sz);
                 _n_reads = 1;
-                read_storage = cast(ubyte*)std.c.stdlib.realloc(read_storage, len);
+                auto realloc_storage = cast(ubyte*)std.c.stdlib.realloc(read_storage, len);
+                if (realloc_storage is null) {
+                    throw new Exception("realloc failed: not enough memory for read");
+                } else {
+                    read_storage = realloc_storage;
+                    max_sz = len;
```

Comment:
`max_sz` was not being updated previously, so the storage would grow but not get used past `max_sz` in future `fill`s.